### PR TITLE
Complete uuids

### PR DESCRIPTION
**Pull request by matbesancon**
Add documentation to the UUID constructor in Base.

Also rewrote `uuid = uuid.value` to `bytes = uuid.value` in the UUID package, it can be confusing to have the UUID struct with the same name as the internal bytes stored as `UInt128`.

### CODE REVIEW

File: base/uuid.jl [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+    Represents a Universally Unique Identifier (UUID).
+    Can be built from a UInt128 (all byte values), from two UInt64 or four UInt32.
+    Conversion from a string is will check the UUID validity.
```
Comment (waldyrious) [inline base/uuid.jl]: Typo: "is will"

Comment (matbesancon) [reply base/uuid.jl]: thanks!

File: base/uuid.jl [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+"""
+    Represents a Universally Unique Identifier (UUID).
+    Can be built from a UInt128 (all byte values), from two UInt64 or four UInt32.
```
Comment (waldyrious) [inline base/uuid.jl]: I'd suggest wrapping the types in backticks, and perhaps tweaking the phrasing to be more consistent when listing the three options -- e.g. "...from one `UInt128` (all byte values), from two `UInt64`, or from four `UInt32`." (or only use "from" once). Note the "a" --> "one", and the added comma after `UInt64`.

Comment (matbesancon) [reply base/uuid.jl]: I just re-read it, agreed on all points

**Review by StefanKarpinski — APPROVED** [review]

**Pull request merged**